### PR TITLE
Warn with two copies of React

### PR DESCRIPTION
**Pull request by spicyj**

### CODE REVIEW

**Comment by spicyj** [comment]
I was gonna just add a comment, but we may as well add a full-blown warning.

**Comment by spicyj** [comment]
Hm, my test passes in jest but fails in grunt test.

**Comment by aaronshaf** [comment]
This would be great!

**Comment by sebmarkbage** [comment]
Fix travis and then ship it?

**Comment by spicyj** [comment]
@sebmarkbage Fixed Travis.

**Comment by chrisirhc** [comment]
Ping, seems like this PR is getting stale. I'm happy to re-roll this into a new fix if this needs some conflict resolution. Hitting this in multiple places.

**Comment by spicyj** [comment]
Couldn't rebase this to have it work at a first attempt. Maybe we work properly on this now? Not sure.

**Comment by JedWatson** [comment]
:grinning: so happy about this - it's going to make so many support issues just disappear. Thanks @spicyj  and everyone!

**Comment by spicyj** [comment]
0.14 also has a few other improvements that should make having multiple copies of React no longer a problem in some cases so hopefully people stop running into these problems.

@JedWatson Thank you.

File: src/renderers/dom/client/__tests__/ReactMount-test.js [hunk 1]
```diff
@@ -199,4 +199,29 @@ describe('ReactMount', function() {
       element.unmount();
     });
   }
+
+  it.only('warns when using two copies of React before throwing', function() {
```
Comment (gaearon) [inline src/renderers/dom/client/__tests__/ReactMount-test.js]: A bit confused, why is `.only` there?

Comment (spicyj) [inline src/renderers/dom/client/__tests__/ReactMount-test.js]: My mistake, thanks.

Comment (spicyj) [inline src/renderers/dom/client/__tests__/ReactMount-test.js]: (#4609)

**Pull request merged**